apps/condo/domains/resident/schema/SuggestServiceProviderService.js

```diff
@@ -0,0 +1,97 @@
+const access = require('@condo/domains/resident/access/SuggestServiceProviderService')
+const { MAX_RESIDENT_SUGGEST_SERVICE_PROVIDER_BY_WINDOW_SEC, RESIDENT_SUGGEST_SERVICE_PROVIDER_WINDOW_SEC } = require('@condo/domains/resident/constants')
+const { RESIDENT } = require('@condo/domains/user/constants/common')
+const { RedisGuard } = require('@condo/domains/user/utils/serverSchema/guards')
+
+const redisGuard = new RedisGuard()
+
+const checkLimits = async (uniqueField) => {
+    await redisGuard.checkCustomLimitCounters(
+        `suggest-service-provider-by-tin-${uniqueField}`,
+        RESIDENT_SUGGEST_SERVICE_PROVIDER_WINDOW_SEC,
+        MAX_RESIDENT_SUGGEST_SERVICE_PROVIDER_BY_WINDOW_SEC,
+    )
+}
+
+const filterOrganizationsByAcquiringContextOrMeterResourceOwnership = async (context, organizations) => {
+    const organizationIds = [...new Set(organizations.map(organization => organization.id))]
+
+    const acquiringIntegrationContexts = await AcquiringIntegrationContext.getAll(context, {
+        organization: { id_in: organizationIds },
+        deletedAt: null,
+        status: CONTEXT_FINISHED_STATUS,
+    })
+
+    const meterResourceOwners = await MeterResourceOwner.getAll(context, {
+        organization: { id_in: organizationIds },
+        deletedAt: null,
+    })
+
+    const organizationsForSuggest = new Set(
+        acquiringIntegrationContexts.map(context => context.organization.id)
+            .concat(meterResourceOwners.map(owner => owner.organization.id))
+    )
+
+    return organizations.filter(organization => organizationsForSuggest.has(organization.id))
+}
+
+
+const SuggestServiceProviderService = new GQLCustomSchema('SuggestServiceProviderByTinService', {
+    types: [
+        {
+            access: true,
+            type: 'input SuggestProviderInput { tinOrName: String! }',
+        },
+        {
+            access: true,
+            type: 'type SuggestProviderOutput { tin: String!, name: String! }',
+        },
+    ],
+    
+    queries: [
+        {
+            access: access.canSuggestProvider,
+            schema: 'suggestProvider (data: SuggestProviderInput!): [SuggestProviderOutput]',
+            resolver: async (parent, args, context = {}) => {
+                if (context.authedItem.type === RESIDENT) {
+                    await checkLimits(context.authedItem.id)
+                }
+
+                const { data: { tinOrName } } = args
+                if (!tinOrName) {
+                    return []
+                }
+
+                const serviceProviders = await Organization.getAll(context, {
+                    OR: [
+                        { tin_starts_with: tinOrName.trim() },
+                        { name_contains: tinOrName.trim() },
+                    ],
+                    deletedAt: null,
+                    type: SERVICE_PROVIDER_TYPE,
+                }, { sortBy: ['name_ASC'] }) // 100 default limit
```

Comment:
Didn't get a comment about 100? where is the limit?

Please pass `first: 100` explicitly. Don't depend on any other logic

Reply:
You should also sort results by tin instead of name

Reply:
We also discussed feature, that organisations with different tin, but the same name must be grouped together.

Lets take 2 organizations:
`{ name: 'my company', tin: '123' }` and `{ name: 'company', tin: '123' }`

By typing "my" in search I should get first one, BUT if it has other aliases (different name, same TIN) it should also be returned.

So mobile clients can group it when register service consumers

Reply:
We now discussed that it's better to stay as it is (returns only unique tins). In suggested case handler should return one record with tin: '123' and any name

Also i don't understand why should we sort results by tin. Analogues have name sort, it is more convenient  for eyes to search from results, that sorted alphabetically. Tin is more like random string\number, so what convenience can it give?